services/server/src/sourcify-chains.ts

```diff
@@ -18,7 +18,30 @@ import dotenv from "dotenv";
 
 dotenv.config();
 
-let sourcifyChainsExtensions: SourcifyChainsExtensionsObject = {};
+// Extended type for FetchRequestRPC with headerEnvName
+export type ExtendedFetchRequestRPC = Omit<FetchRequestRPC, "headers"> & {
+  headers?: Array<{
+    headerName: string;
+    headerValue?: string;
+    headerEnvName?: string;
+  }>;
+};
+
+// Extended type for SourcifyChainsExtensionsObject that uses ExtendedFetchRequestRPC
+export interface ExtendedSourcifyChainsExtensionsObject {
+  [chainId: string]: {
+    sourcifyName: string;
+    supported: boolean;
+    etherscanApi?: {
+      apiURL: string;
+      apiKeyEnvName?: string;
+    };
+    fetchContractCreationTxUsing?: any; // Using any to avoid importing the full type
+    rpc?: Array<string | BaseRPC | APIKeyRPC | ExtendedFetchRequestRPC>;
+  };
```

Comment:
You don't need to redeclare the whole type here. You can just overwrite the `rpc` property of `SourcifyChainExtension`.